mlflow/tensorflow.py

```diff
@@ -293,15 +293,27 @@ def save_model(
         _save_example(mlflow_model, input_example, path)
     root_relative_path = _copy_file_or_tree(src=tf_saved_model_dir, dst=path, dst_dir=None)
     model_dir_subpath = "tfmodel"
-    shutil.move(os.path.join(path, root_relative_path), os.path.join(path, model_dir_subpath))
+    model_dir_path = os.path.join(path, model_dir_subpath)
+    shutil.move(os.path.join(path, root_relative_path), model_dir_path)
 
-    conda_env, pip_requirements, pip_constraints = (
-        _process_pip_requirements(
-            get_default_pip_requirements(), pip_requirements, extra_pip_requirements,
-        )
-        if conda_env is None
-        else _process_conda_env(conda_env)
+    flavor_conf = dict(
+        saved_model_dir=model_dir_subpath,
+        meta_graph_tags=tf_meta_graph_tags,
+        signature_def_key=tf_signature_def_key,
     )
+    if conda_env is None:
+        default_reqs = get_default_pip_requirements()
+        if pip_requirements is None:
+            save_path = os.path.join(path, MLMODEL_FILE_NAME)
+            Model().add_flavor(FLAVOR_NAME, **flavor_conf).save(save_path)
+            inferred_reqs = mlflow.infer_pip_requirements(path, FLAVOR_NAME, fallback=default_reqs,)
+            os.remove(save_path)
```

Comment:
Can we make the same adjustment recommended here? https://github.com/mlflow/mlflow/pull/4518/files#r683751938. Can we also add an inline comment?